ports/esp32/boards/UM_FEATHERS3/modules/feathers3.py

```diff
@@ -0,0 +1,93 @@
+# FeatherS3 MicroPython Helper Library
+# 2022 Seon Rozenblum - Unexpected Maker
+#
+# Project home:
+# http://feathers3.io
+
+# Import required libraries
+from micropython import const
+from machine import Pin, ADC
+import time
+
+# FeatherS3 Hardware Pin Assignments
+
+# Sense Pins
+VBUS_SENSE = const(34)
+VBAT_SENSE = const(2)
+
+# RGB LED, LDO2 & Other Pins
+RGB_DATA = const(40)
+LDO2 = const(39)
+LED = const(13)
+AMB_LIGHT = const(4)
+
+# SPI
+SPI_MOSI = const(35)
+SPI_MISO = const(37)
+SPI_CLK = const(36)
+
+# I2C
+I2C_SDA = const(8)
+I2C_SCL = const(9)
+
+# Helper functions
+
+# LED & Ambient Light Sensor control
+def led_set(state):
+    """Set the state of the BLUE LED on IO13"""
+    l = Pin(LED, Pin.OUT)
+    l.value(state)
+
+
+def led_blink():
+    """Toggle the BLUE LED on IO13"""
+    l = Pin(LED, Pin.OUT)
+    l.value(not l.value())
+
+
+# Create ADC and set attenuation and return the ambient light value from the onboard sensor
+def get_amb_light():
+    """Get Ambient Light Sensor reading"""
+    adc = ADC(Pin(AMB_LIGHT))
+    adc.atten(ADC.ATTN_11DB)
+    return adc.read()
+
+
+def set_ldo2_power(state):
+    """Enable or Disable power to the second LDO"""
+    Pin(LDO2, Pin.OUT).value(state)
+
+
+def get_battery_voltage():
+    """
+    Returns the current battery voltage. If no battery is connected, returns 4.2V which is the charge voltage
+    This is an approximation only, but useful to detect if the charge state of the battery is getting low.
+    """
+    adc = ADC(Pin(VBAT_SENSE))  # Assign the ADC pin to read
+    # We are going to read the ADC 10 times and discard the results as we can't guarantee the ADC is calibrated or stable yet after boot. Not sure why we have to do this :(
+    for _ in range(10):
+        adc.read()
```

Comment:
There is now an `adc.read_uv()` method which gives calibrated readings in microvolts.  That might be better to use here?